src/controller/SetUpCodePairer.cpp

```diff
@@ -448,6 +448,9 @@ void SetUpCodePairer::OnPairingComplete(CHIP_ERROR error)
         return;
     }
 
+    if (pairingDelegate != nullptr && pairingDelegate->IsDiscoverOnce())
```

Comment:
This will not call OnPairingComplete(error) anymore. Why is that ok?

Reply:
I am generally unclear what we are trying to bypass: TryNext... or OnPairingComplete. It seems to me like OnPairingComplete should be called whenever it actually completes and eliding it seems wrong.

Reply:
fixed.